### PR TITLE
[AF-1555]: Upgrade to Wildfly14

### PR DESCRIPTION
**Pull request by adrielparedes**
## Description

Upgrate to Wildfly 14

There is a concern related to SNAPSHOT version that will be solved in jboss-integration-platform-bom

@ederign would you mind to review?

Here is kie-wb, https://www.dropbox.com/sh/54ngqqacaqjh2f9/AADoeCmukZ7xytqTpYJCTfEma?dl=0

## Related to:
* https://github.com/jboss-integration/jboss-integration-platform-bom/pull/427
* https://github.com/errai/errai/pull/363
* https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/843
* https://github.com/kiegroup/appformer/pull/483
* https://github.com/kiegroup/kie-wb-common/pull/2147
* https://github.com/kiegroup/jbpm/pull/1336
* https://github.com/kiegroup/jbpm-wb/pull/1206
* https://github.com/kiegroup/drools/pull/2086
* https://github.com/kiegroup/droolsjbpm-integration/pull/1583
* https://github.com/kiegroup/kie-wb-distributions/pull/821 
* https://github.com/kiegroup/kie-docs/pull/1108
* https://github.com/kiegroup/optaplanner/pull/437
* https://github.com/kiegroup/optaplanner-wb/pull/304

### CODE REVIEW

**Comment by ederign** [comment]
@jhrcek could you please assign this for someone from QE?

File: pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@
     <groupId>org.jboss.integration-platform</groupId>
     <artifactId>jboss-integration-platform-bom</artifactId>
     <!-- Keep in sync with the parent version of uberfire-bom/pom.xml -->
-    <version>8.1.0.Final</version>
+    <version>8.2.0-SNAPSHOT</version>
```
Comment (jhrcek) [inline pom.xml]: IP BOM Snapshot version again..

File: pom.xml [hunk 2]
```diff
@@ -139,7 +139,7 @@
 
     <!-- Required since support for ELS 2.x. Keep in sync with kie-parent or remove when those
           two versions are being updated on the IP BOM.-->
-    <version.com.fasterxml.jackson>2.6.2</version.com.fasterxml.jackson>
+    <version.com.fasterxml.jackson>2.9.5</version.com.fasterxml.jackson>
```
Comment (jhrcek) [inline pom.xml]: The same as above:
There seems to be no reason to keep 2 properties for jackson version in this file now that both versions are aligned.
Please remove one of them and replace occurences with the other.

**Comment by jhrcek** [comment]
@ederign @adrielparedes I took a look and I can test it once all the comments are addressed (most importantly there should be no dependence on IP BOM snapshot versions)

**Comment by adrielparedes** [comment]
@jhrcek you are right about jackson dep, removing it. About Snapshot version, with @mbiarnes we agreed that for now it will be snapshot after a build he has goes green. After that we will plan and update those version to Final.

**Comment by jhrcek** [comment]
@adrielparedes @ederign At this point it's not clear to me what is expected from QE.
In particular QE officially doesn't test the binaries in any way on Wildfly 14.
Do you want me to build the PR ensemble locally and try out that kie-wb can be deployed on wildfly?
Also please let me know when you think it's ready for review / testing / when it's "ready to be checked" (and ideally what you'd expect me to check).

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by adrielparedes** [comment]
jenkins retest this

**Comment by dgutierr** [comment]
@jhrcek From my understanding, what's need to be tested is the full build against the latest available EAP 7.2 beta.

**Comment by jhrcek** [comment]
@dgutierr got it. Once this PR is merged, QE will switch to that EAP version in our automation (unfortunately we don't yet have a way to run all the tests based on just the PRs)

**Comment by ederign** [comment]
jenkins execute full downstream build

**Comment by adrielparedes** [comment]
jenkins execute full downstream build

**Pull request merged**